hazelcast/src/main/java/com/hazelcast/query/impl/predicates/EvaluateVisitor.java

```diff
@@ -51,7 +51,7 @@ public Predicate visit(AndPredicate andPredicate, Indexes indexes) {
 
             EvaluatePredicate evaluatePredicate = (EvaluatePredicate) subPredicate;
             String indexName = evaluatePredicate.getIndexName();
-            Index index = indexes.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME);
+            Index index = indexes.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME, -1);
```

Comment:
I am a little worried with plain `-1` in many places. Shouldn't we declare it as a `static final` constant instead?